x/mongo/driver/auth/mongodbaws.go

```diff
@@ -35,11 +36,17 @@ type MongoDBAWSAuthenticator struct {
 
 // Auth authenticates the connection.
 func (a *MongoDBAWSAuthenticator) Auth(ctx context.Context, cfg *Config) error {
+	httpClient := cfg.HTTPClient
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+		defer httpClient.CloseIdleConnections()
```

Comment:
Similar to previous comment: I'm concerned about calling `CloseIdleConnections` on the package-global `http.DefaultClient` because it may interact unexpectedly with other code using `http.DefaultClient`. Can we create a new `http.Client` here instead, or use the `http.Client` from the `mongo.Client`?

Reply:
see other comment